rust/chains/hyperlane-ethereum/src/mailbox.rs

```diff
@@ -130,11 +130,16 @@ where
     }
 
     #[instrument(err, skip(self))]
-    async fn fetch_logs(
-        &self,
-        from: u32,
-        to: u32,
-    ) -> ChainResult<Vec<(HyperlaneMessage, LogMeta)>> {
+    async fn fetch_logs(&self, range: IndexRange) -> ChainResult<Vec<(HyperlaneMessage, LogMeta)>> {
+        let (from, to) = match range {
+            IndexRange::Blocks(from, to) => (from, to),
```

Comment:
This is more of a style thing but if the ethereum indexer only supports `IndexRange::Blocks`, it would be more maintainable to replace the let-match with a `let IndexRange::Blocks(from, to) else { ... }`, because new enum items wouldn't break this. This pattern could be applied in the sealevel mailbox as well